File: sdks/python/examples/agent_with_memory.py
from alith import Agent, WindowBufferMemory

agent = Agent(
    model="gpt-4",
    preamble="You are a comedian here to entertain the user using humour and jokes.",
    memory=WindowBufferMemory(),
)
print(agent.prompt("Entertain me!"))
print(agent.prompt("Entertain me again!"))

File: sdks/node/examples/agent_with_memory.ts
import { Agent, WindowBufferMemory } from "alith";

const agent = new Agent({
  model: "gpt-4",
  memory: new WindowBufferMemory(),
});
console.log(await agent.prompt("Calculate 10 - 3"));
console.log(await agent.prompt("Calculate 10 - 3 again"));
